### PR TITLE
Update poetry publish workflow version to latest

### PR DESCRIPTION
**Pull request by raminqaf**
image

There is a way to increase the timeout:
https://python-poetry.org/docs/faq/#my-requests-are-timing-out

### Edit
The above issue is still not solved. Even after setting the env variable, the timeout does not increase. Related issue for TestPyPI: https://github.com/pypi/support/issues/3396

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -6,9 +6,8 @@ on:
 
 jobs:
   publish:
-    uses: bakdata/ci-templates/.github/workflows/python-poetry-publish-pypi.yaml@1.25.3
+    uses: bakdata/ci-templates/.github/workflows/python-poetry-publish-pypi.yaml@1.40.4
     with:
       publish-to-test: false
-      poetry-version: "1.5.1"
```
[the remark this answers is not in the document]
Comment (raminqaf) [reply .github/workflows/publish.yaml:12]: done

File: .github/workflows/publish.yaml [hunk 2]
```diff
@@ -9,5 +9,6 @@ jobs:
     uses: bakdata/ci-templates/.github/workflows/python-poetry-publish-pypi.yaml@1.40.4
     with:
       publish-to-test: false
+      poetry-version: "1.4.1"
```
Comment (disrupted) [inline .github/workflows/publish.yaml:12]: I just saw the timeout env var was only added in Poetry 1.5.0. https://github.com/python-poetry/poetry/pull/7081
Does it make sense to update to the latest version?

Comment (raminqaf) [reply .github/workflows/publish.yaml:12]: It make sense to have that above 1.5.0. But we still have the issue with [poetry and windows](https://github.com/bakdata/kpops/pull/430/files#diff-944291df2c9c06359d37cc8833d182d705c9e8c3108e7cfe132d61a06e9133ddL31). I can give it a try with the newest version (1.7.1)

Comment (raminqaf) [reply .github/workflows/publish.yaml]: It seems that the problem with the windows runners is not occurring. We can keep the 1.7.1

**Pull request merged**